main.go

```diff
@@ -95,6 +96,17 @@ func main() {
 		}
 	}
 
+	// set timeouts for kube client
+	kubeConfig.Transport = &http.Transport{
+		Dial: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+		}).Dial,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ResponseHeaderTimeout: 10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+	}
```

Comment:
https://github.com/zalando/skipper/blob/master/dataclients/kubernetes/kube.go#L299-L326
Looks good to me, but I think you should add the goroutine that closes idle connections in case this transport hast to do another DNS lookup for whatever reason. https://github.com/golang/go/issues/23427

Reply:
For this specific case it's not really needed since the controller just goes to `10.3.0.1`.